src/test/scala/com/github/agourlay/cornichon/examples/superHeroes/server/models.scala

```diff
@@ -10,24 +10,41 @@ case class Publisher(name: String, foundationYear: Int, location: String)
 
 case class SuperHero(name: String, realName: String, city: String, hasSuperpowers: Boolean, publisher: Publisher)
 
+import sangria.schema._
+import sangria.marshalling.sprayJson._
+
 object GraphQlSchema {
+  import JsonSupport._
 
   implicit val PublisherType = deriveObjectType[Unit, Publisher](
-    ObjectTypeName("Publisher"),
     ObjectTypeDescription("A comics publisher.")
   )
 
   implicit val SuperHeroType = deriveObjectType[Unit, SuperHero](
-    ObjectTypeName("Superhero"),
     ObjectTypeDescription("A superhero.")
   )
 
+  implicit val PublisherInputType = deriveInputObjectType[Publisher](
+    InputObjectTypeName("PublisherInput")
+  )
+
+  implicit val SuperHeroInputType = deriveInputObjectType[SuperHero](
+    InputObjectTypeName("SuperHeroInput")
+  )
+
   val QueryType = deriveObjectType[Unit, TestData](
     ObjectTypeName("Root"),
-    ObjectTypeDescription("Gateway to awesomeness.")
+    ObjectTypeDescription("Gateway to awesomeness."),
+    ExcludeFields("updateSuperhero")
+  )
+
+  val MutationType = deriveObjectType[Unit, TestData](
+    ObjectTypeName("RootMut"),
+    ObjectTypeDescription("Gateway to mutation awesomeness!"),
+    ExcludeFields("publisherByName", "superheroByName")
```

Comment:
I just realized, instead of using an `@GraphQLField` annotation, we can use `IncludeMethods` setting.